controllers/applicationsController/deleteSingleApplication.js

```diff
@@ -10,7 +11,11 @@ const deleteSingleApplication = async (req, res, next) => {
     if (!application) {
       return next(new CustomError(400, "Application not found"));
     }
-    if (application.organizationId !== organizationId) {
+    if (
+      !mongoose.Types.ObjectId(application.organizationId).equals(
+        mongoose.Types.ObjectId(organizationId)
+      )
+    ) {
       return next(
         new CustomError(403, "You're not allowed to access this resource")
```

Comment:
We should be actually using a 401, because this is an authorization issue. The endpoint itself is not forbidden.